### PR TITLE
Pull config from semgrep app

### PR DESCRIPTION
**Pull request by underyx**
Merge https://github.com/returntocorp/semgrep-app/pull/44 before

### CODE REVIEW

File: semgrep_agent/main.py [hunk 1]
```diff
@@ -80,8 +69,36 @@ def main(
...
+                Please either set a config in the action's configuration according to
+                https://github.com/returntocorp/semgrep-action#configuration
+                or commit your own rules at the default path of .bento/semgrep.yml
+            """
+            message = dedent(message).strip()
+            click.echo(message, err=True)
+
     obj.sapp.report_start()
 
+    if not config and not (Path(".bento") / "semgrep.yml").is_file():
```
Comment (nbrahms) [inline semgrep_agent/main.py:74]: Why are we doing this twice?

File: semgrep_agent/semgrep_app.py [hunk 2]
```diff
@@ -45,6 +47,29 @@ def report_start(self) -> None:
         else:
             self.scan_id = response.json()["scan"]["id"]
 
+    def fetch_rules_text(self) -> str:
+        """Get a YAML string with the configured semgrep rules in it."""
+        if self.scan_id is None:
+            raise RuntimeError("sapp is configured so we should've gotten a scan_id")
```
Comment (nbrahms) [inline semgrep_agent/semgrep_app.py:53]: sapp?

Comment (underyx) [reply semgrep_agent/semgrep_app.py:53]: `Sapp` is the client class and it's at `ctx.obj.sapp`. This is just here to appease mypy, so I thought I'd use the dev-facing name. I could rename it to `semgrep_app` everywhere?

File: semgrep_agent/main.py (hunk 1 above)
Comment (underyx) [reply semgrep_agent/main.py:74]: Ohhh, good catch. Just a bad merge conflict resolution.

File: semgrep_agent/semgrep_app.py (hunk 2 above)
Comment (nbrahms) [reply semgrep_agent/semgrep_app.py:53]: nah it's cool, I was just confused

**Review by nbrahms — APPROVED** [review]

**Review by nbrahms — APPROVED** [review]

**Pull request merged**